next-packages/utils/src/index.js

```diff
@@ -41,17 +53,20 @@ export const insertStyles = (
       // the registered cache to know whether a style is global or not
       // also, note that this check will be dead code eliminated in the browser
       (isBrowser === false && context.compat !== undefined)) &&
-    context.registered[`${context.key}-${insertable.name}`] === undefined
+    context.registered[className] === undefined
   ) {
-    context.registered[`${context.key}-${insertable.name}`] = insertable.styles
+    context.registered[className] = insertable.styles
   }
   if (context.inserted[insertable.name] === undefined) {
-    let rules = context.stylis(
-      `.${context.key}-${insertable.name}`,
-      insertable.styles
-    )
+    let rules = context.stylis(`.${className}`, insertable.styles)
     context.inserted[insertable.name] = true
 
+    if (insertable.map !== undefined && process.env.NODE_ENV !== 'production') {
```

Comment:
It’s better the way it is since the check is pure and cheaper than checking process.env in node

Reply:
hm, it think (not sure) that this check is pure only if one uses `pure_getters` or similar option, if you are concerned with process.env implications maybe it would be sufficient to cache it upfront and use the cached value here?